hibernate-core/src/main/java/org/hibernate/boot/internal/AttributeConverterDescriptorImpl.java

```diff
@@ -150,7 +150,7 @@ else if ( member instanceof Field ) {
 
 	private static Method memberMethod;
 
-	private static Member toMember(XProperty xProperty) {
+	private static synchronized Member toMember(XProperty xProperty) {
```

Comment:
Good catch, something is fishy here! Yet synchronizing is not the best solution, as it would introduce a contention point while it's not necessary.
The `memberMethod` should be made `final` and initialized statically.

Also, would be nice to have fields at the beginning of the class.